src/track/track.cpp

```diff
@@ -989,6 +989,63 @@ void Track::shiftCuePositionsMillis(double milliseconds) {
     markDirtyAndUnlock(&locked);
 }
 
+void Track::setHotcueIndicesSortedByPosition(bool compress) {
```

Comment:
Even when there are only two options, I would prefer to make this an enum, not a bool.  This makes it more clear at the call site what the parameter is for.  (Also I would like to add my own scheme, assigning numbers by relative track position, which would be a third option here)

Reply:
that would be a post-merge change

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#i4-make-interfaces-precisely-and-strongly-typed

Reply:
I made it an enum class and rebased.
Let me know when you want to start reviewing, I'll stop force-pushing then.